gadgets/trace_tcpconnect/test/trace_tcpconnect_test.go

```diff
@@ -0,0 +1,130 @@
+
+	Timestamp string `json:"timestamp"`
+	MntNsID   uint64 `json:"mntns_id"`
+
+	Comm string `json:"comm"`
+	Pid  uint32 `json:"pid"`
+	Tid  uint32 `json:"tid"`
+	Uid  uint32 `json:"uid"`
+	Gid  uint32 `json:"gid"`
+
+	Latency     uint64           `json:"latency,omitempty"`
+	SrcEndpoint utils.L4Endpoint `json:"src"`
+	DstEndpoint utils.L4Endpoint `json:"dst"`
+	ErrorRaw    uint32           `json:"error_raw"`
+}
+
+func TestTraceTcpconnect(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-trace-tcpconnect"
+	containerImage := "docker.io/library/nginx:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-trace-tcpconnect")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+	cmds := "nginx && while true; do curl 127.0.0.1; sleep 0.1; done"
+	testContainer := containerFactory.NewContainer(containerName, cmds, containerOpts...)
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime)))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns)))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntries := &traceTcpconnectEvent{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				SrcEndpoint: utils.L4Endpoint{
+					Addr:    "127.0.0.1",
+					Version: 4,
+					Port:    utils.NormalizedInt,
+					Proto:   6,
+				},
+				DstEndpoint: utils.L4Endpoint{
+					Addr:    "127.0.0.1",
+					Version: 4,
+					Port:    utils.NormalizedInt,
+					Proto:   6,
+				},
+				Comm:     "curl",
+				ErrorRaw: 0,
+
+				Uid:       utils.NormalizedInt,
+				Gid:       utils.NormalizedInt,
+				MntNsID:   utils.NormalizedInt,
+				Timestamp: utils.NormalizedStr,
+				Pid:       utils.NormalizedInt,
+				Tid:       utils.NormalizedInt,
+			}
+
+			normalize := func(e *traceTcpconnectEvent) {
+				utils.NormalizeCommonData(&e.CommonData)
+				utils.NormalizeInt(&e.MntNsID)
+				utils.NormalizeString(&e.Timestamp)
+				utils.NormalizeInt(&e.Pid)
+				utils.NormalizeInt(&e.Tid)
```

Comment:
Normalizing `Uid` and `Gid` is needed

Reply:
@burak-ok so should i also normalize the Latency to check its existence since it depends on the --latency flag??

Reply:
Which also means that we have to do the same for `uid` and `gid`